ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py

```diff
@@ -32,7 +32,7 @@ def router_bgp(self) -> dict | None:
                 continue
 
             if p2p_link["data"]["bgp_as"] is None or p2p_link["data"]["peer_bgp_as"] is None:
-                raise AristaAvdMissingVariableError("l3_edge.p2p_links.[].as")
+                raise AristaAvdMissingVariableError(f"{self.data_model}.p2p_links.[].as or {self.data_model}.p2p_links_profiles.[].as")
```

Comment:
Can we have better error message. Like add index of p2p_links. 
Also if i set only one as then getting below error
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: IndexError: list index out of range
fatal: [core-4-multicast -> localhost]: FAILED! => {"changed": false, "msg": "list index out of range"}
```

Reply:
Fixed the index out of range error.
For error message we can not get the `index` in the current code implementation. So keeping it like this.